torch/csrc/jit/python/pybind_utils.h

```diff
@@ -571,7 +571,11 @@ inline std::string friendlyTypeName(py::handle obj) {
     ss << "))";
     return ss.str();
   } else {
-    return py::str(obj.get_type().attr("__name__"));
+    auto match = tryToInferType(obj);
+    if (!match.success()) {
+      return py::str(obj.get_type().attr("__name__"));
+    }
+    return toIValue(obj, match.type()).type()->annotation_str();
```

Comment:
I am probably missing something, why is this change needed? Since `Union` doesn't have a corresponding IValue tag, I am not sure how these new code relate to `Union`

Reply:
Reverted